### ISSUE TITLE
right place to start WifiStation.waitConnection

### ISSUE THREAD

**Issue by avr39-ripe** [SmingHub/Sming#504]
Hello! After some SDK change WifiStation.waitConnection do not work for me as expected - if I invoce it in such order
```c++
WifiStation.enable(true);
WifiStation.config(ActiveConfig.StaSSID, ActiveConfig.StaPassword);
WifiStation.waitConnection(StaConnectOk, StaConnectTimeout, StaConnectFail);
```
and if esp can't connect to ActiveConfig.StaSSID wifi network method StaConnectFail never called. Instead of this there are meny "...recconect after 1 s" messages than came forever. If network is reachable, and esp8266 successfuly conects to it all works as expected and StaConnectOk called.
After some experiments and investigation I found that if I just change the order ov functions call all works as expected in BOTH cases: either accessible or inaccessible wifi network. So here is correct order:
```c++
WifiStation.waitConnection(StaConnectOk, StaConnectTimeout, StaConnectFail);
WifiStation.enable(true);
WifiStation.config(ActiveConfig.StaSSID, ActiveConfig.StaPassword);
```
Do please comment on this, test, confirm or suggest better/cleander solution.

**Comment by tprochazka** [comment]
Just small note. You don't need to call
WifiStation.config(ActiveConfig.StaSSID, ActiveConfig.StaPassword);
in every time in init(), Espressif SDK remember last value itself.

**Comment by avr39-ripe** [comment]
This will be resolved by #611 which in fact changes the way of handling wifi related events.
